ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -0,0 +1,294 @@
+        report::hybrid::{HybridReport, IndistinguishableHybridReport},
+        secret_sharing::IntoShares,
+        test_fixture::{
+            hybrid::TestHybridRecord, RoundRobinInputDistribution, TestWorld, TestWorldConfig,
+            WithShards,
+        },
+    };
+
+    #[tokio::test]
+    async fn hybrid_oprf() {
+        let world: TestWorld<WithShards<SHARDS, RoundRobinInputDistribution>> =
+            TestWorld::with_shards(TestWorldConfig::default());
+
+        let contexts = world.contexts();
+
+        const SHARDS: usize = 2;
+
+        let records = vec![
+            TestHybridRecord::TestImpression {
+                match_key: 12345,
+                breakdown_key: 2,
+            },
+            TestHybridRecord::TestImpression {
+                match_key: 68362,
+                breakdown_key: 1,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 12345,
+                value: 5,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 68362,
+                value: 2,
+            },
+            TestHybridRecord::TestImpression {
+                match_key: 68362,
+                breakdown_key: 1,
+            },
+            TestHybridRecord::TestConversion {
+                match_key: 68362,
+                value: 7,
+            },
+        ];
+        let indices_to_compare = [(0, 2), (1, 3), (1, 4), (1, 5)];
+
+        let shares: [Vec<HybridReport<BA8, BA3>>; 3] = records.iter().cloned().share();
+
+        let indistinguishable_reports: [Vec<IndistinguishableHybridReport<BA8, BA3>>; 3] = shares
+            .iter()
+            .map(|s| s.into_iter().map(|r| r.clone().into()).collect::<Vec<_>>())
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("Expected exactly 3 elements");
+
+        let chunked_reports: [Vec<Vec<IndistinguishableHybridReport<BA8, BA3>>>; 3] =
+            indistinguishable_reports
+                .iter()
+                .map(|vec| {
+                    let mid = vec.len() / SHARDS;
+                    vec.chunks(mid)
+                        .map(|chunk| chunk.to_vec())
+                        .collect::<Vec<_>>()
+                })
+                .collect::<Vec<_>>()
+                .try_into()
+                .expect("Expected exactly 3 elements");
+
+        // #[allow(clippy::large_futures)]
+        // let results = chunked_reports
+        //     .into_iter()
+        //     .zip(contexts)
+        //     .map(|(reports_by_helper, helper_ctxs)| {
+        //         reports_by_helper
+        //             .into_iter()
+        //             .zip(helper_ctxs)
+        //             .map(|(reports, ctx)| async move {
+        //                 compute_prf_for_inputs(ctx, &reports).await.unwrap()
+        //             })
+        //             .collect::<Vec<_>>()
+        //     })
+        //     .collect::<Vec<_>>();
```

Comment:
I can't seem to get this to work. the loop approach below seems to work. it's a bit less ergonomic, but hopefully it's acceptable for a test.

Reply:
I'll delete this if we're ok with the approach below.